portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,7 +81,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       BlogHashMap blogMap = new BlogHashMap();
       blogMap.putInMap(messages);
 
-      // Load messages from BlogHashMap and respond with gson.
+      // If (user loads all BlogMessages) 
+      LinkedList<BlogMessage> allBlogMessages = blogMap.getMessages();
+
+      if (numberOfCommentsToDisplay < 1 || numberOfCommentsToDisplay >= allBlogMessages.size()) {
```

Comment:
From a usability standpoint, I think it would be better to set numberOfCommentsToDisplay = allBlogMessages.size() if the user is requesting more comments than are available. The user is likely to want the max number of comments anywas, and this saves the user a round trip of input and response.